src/averageAges.js

```diff
@@ -55,7 +65,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const peopleHaveMother = calcAvarageAgeDiff(people, false);
+  const sumAllAgesDiff = peopleHaveMother.reduce((sum, e) => sum + e, 0);
+  const menHaveMother = calcAvarageAgeDiff(people, true);
+  const sumSonsAgesDiff = menHaveMother.reduce((sum, e) => sum + e, 0);
+
+  return onlyWithSon
+    ? sumSonsAgesDiff / menHaveMother.length
+    : sumAllAgesDiff / peopleHaveMother.length;
+}
+
+function calcAvarageAgeDiff(arr, son) {
+  let result = arr.filter(kid => arr.some(mom => mom.name === kid.mother));
+
+  if (son) {
+    result = result.filter(kid => kid.sex === 'm');
+  }
```

Comment:
try to avoid `if`
![image](https://user-images.githubusercontent.com/70009205/216370220-4db2cd26-8129-49e0-a3f1-b9bf058cd422.png)